apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -140,58 +139,6 @@ func (v *validator) validateServiceAccount(ctx context.Context, tempo *Microserv
 	return allErrs
 }
 
-func (v *validator) validateStorageSecret(tempo *Microservices, storageSecret *corev1.Secret) field.ErrorList {
```

Comment:
Should we consider doing validation here as well and to catch "init errors"? 

I find it useful to inform the user that the config is not correct when creating the instance

Reply:
Can we guarantee that users always create the secret before the tempo CR?

An additional feature of this PR is that now users can apply the manifests in any order and it'll still work, while before the order was important (first apply the secret, then the CR). That's also how Pods works, if they reference a secret which doesn't exist the pod can still get created, but the Pod's status is `CreateContainerConfigError` and the `ContainersReady` condition is set to false.

Reply:
> Can we guarantee that users always create the secret before the tempo CR?

Good comment. I definitely like the implementation in this PR. However, I would still advise users to create the secret first. The error propagation in the validating webhook is just better in my opinion.